lua/neo-tree/sources/common/commands.lua

```diff
@@ -152,7 +152,10 @@ M.close_node = function(state, callback)
     target_node = parent_node
   end
 
-  if target_node and target_node:has_children() then
+  local root = tree:get_nodes()[1]
```

Comment:
I'm not sure if this is the best way to determine if it the root or not. What happens when the hide root config is used? Is the root still in this collection at position 1 or could that be a normal folder that should be collapsible?

Reply:
I have tested the code with the `hide_root_node` config set, and things work well. According to the [nui's documentation](https://github.com/MunifTanjim/nui.nvim/tree/main/lua/nui/tree#treeget_nodes), `get_nodes` returns a list of roots of the tree, and in our case it is a singular list containing the directory's root.

If I'm not wrong then `hide_root_node` is only used [here](https://github.com/nvim-neo-tree/neo-tree.nvim/blob/main/lua/neo-tree/ui/renderer.lua#L1128), which only prevent the root node from being rendered from the screen. The tree itself is never modified, so I believe the `root` node should remain the same.

Reply:
OK, just wanted to make sure that situation was tested. Thanks for the explanation.